keyboards/anne_pro/anne_pro_lighting.c

```diff
@@ -192,3 +196,35 @@ void anne_pro_lighting_set_keys(uint8_t keys, uint8_t *payload) {
     uart_tx_ringbuf_write(&led_uart_ringbuf, 5, buf);
     uart_tx_ringbuf_write(&led_uart_ringbuf, 5 * keys, payload);
 }
+
+/* Set Caps Lock red led on */
+void anne_pro_lighting_caps_lock_on(){
+		if(!leds_enabled && !not_init){
+			/* Wake up the LED controller if off */
+			writePinHigh(C15);
+			chThdSleepMilliseconds(50);
+			not_init=true;
+		}
+		if(leds_enabled) // solves problems when LT(1,KC_CAPS) is used in keymap(Tap = caps, Hold = MO(1))
+			anne_pro_lighting_update();
```

Comment:
`anne_pro_lighting_update` only sends out data from the buffer, why is that needed here?

Reply:
There is a problem, could not find out the origin, that when T(1,KC_CAPS) is used instead of KC_CAPS, it would call `anne_pro_lighting_caps_lock_on` execute it but did not transmit the payload , so i thought that mb the buffer was busy and that by  freeing it before doing the tx would work, and worked!!
this problem only happens when the lights are on. As you know how the uart and the ringbuffer works you might be able to find out why.

Reply:
not needed anymore